### PR TITLE
[READY] Adds selector lists to SDQL

### PR DESCRIPTION
**Pull request by JJRcop**
## About The Pull Request

I've added a selector array query. It's a special construct formed like this:
```bnf
'@[' <selectors> ']'
```
The `<selectors>` is the very same way that you normally use to select datums for all of the queries.
```bnf
UPDATE <selectors> SET <assignments>
SELECT <selectors>
```

I chose `@[ ]` because it's adding an @ sign to the existing SDQL list symbols, since it still generates a list.

## Why It's Good For The Game

It adds more flexibility to SDQL, allowing admins to do more with the language. I find it very convenient in testing to really narrow down the search to say, an /area, so SDQL will only look for datums in that /area's .contents, when paired with MAP.

query
result

## Changelog
:cl: JJRcop
add: SDQL2 now features a list constructor that takes a selector query. 
tweak: (sub expressions) will work with var access, like .vars and [list indexes]
/:cl:

i stayed up too late

### CODE REVIEW

**Comment by optimumtact** [comment]
dude just revive NTSL and give it the same hooks for admins

it's a lot easier to think about imperatively

**Comment by JJRcop** [comment]
I'm not very interested in reviving NTSL right now. While it would be much better than SDQL, I would have to do so much more work than this, including research on making/maintaining a programming language and its various systems, along with the specific language attributes NTSL has.

It might not be as hard as I'm thinking it is, but I don't have the desire right now.

File: code/modules/admin/verbs/SDQL2/SDQL_2.dm [hunk 1]
```diff
@@ -535,6 +539,8 @@ GLOBAL_DATUM_INIT(sdql2_vv_statobj, /obj/effect/statclick/SDQL2_VV_all, new(null
 					objs[j] = SDQL_expression(x, expression)
 					SDQL2_TICK_CHECK
 					SDQL2_HALT_CHECK
+				if(length(objs) == 1 && islist(objs[1]))
```
Comment (AnturK) [inline code/modules/admin/verbs/SDQL2/SDQL_2.dm]: What's with conditional unwrapping, this is going to be an issue if anyone wants to retrieve list variables. This should probably be controlled by the user.

Comment (JJRcop) [reply code/modules/admin/verbs/SDQL2/SDQL_2.dm]: This is to get around an issue where defining a list then immediately accessing it doesn't work. I'll see what I can do about reverting this and fixing the issue itself.

Comment (JJRcop) [reply code/modules/admin/verbs/SDQL2/SDQL_2.dm]: I've reverted the unwrapping and allowed variable to process expressions now. The proper solution to my above problem is `(@[/area/crew_quarters/bar MAP contents])[1]`

**Comment by JJRcop** [comment]
Resolved @AnturK's concerns.

**Pull request merged**